hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -101,6 +102,16 @@ public CleanPlanner(HoodieEngineContext context, HoodieTable<T, I, K, O> hoodieT
     this.fgIdToPendingLogCompactionOperations = fileSystemView.getPendingLogCompactionOperations()
         .map(entry -> Pair.of(new HoodieFileGroupId(entry.getValue().getPartitionPath(), entry.getValue().getFileId()), entry.getValue()))
         .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+
+    // load all partitions in advance if necessary.
+    if (config.isMetadataTableEnabled()
+        && config.getCleanerPolicy().equals(HoodieCleaningPolicy.KEEP_LATEST_FILE_VERSIONS)
+        && (config.getViewStorageConfig().getStorageType().equals(FileSystemViewStorageType.MEMORY)
+            || (config.getViewStorageConfig().getStorageType().equals(FileSystemViewStorageType.REMOTE_FIRST)
+                && config.getViewStorageConfig().getSecondaryStorageType().equals(FileSystemViewStorageType.MEMORY)))) {
+      LOG.info("Load all partitions and files into file system view in advance when using KEEP_LATEST_FILE_VERSIONS.");
+      fileSystemView.loadAllPartitions();
```

Comment:
Will this approach be slower than master branch implementation in spark?  In spark engine,  partitions will be loaded parallelly in subtasks while here partitions will be loaded by a single thread.

Reply:
Thanks @stream2000 for your attention.  I will update our test result later, in summary
Data size: 300T
Partition numbers: 63,000
KEEP_LATEST_FILE_VERSIONS

|  feature   | clean time |
|  ----  | ----  |
| Disable MDT | 2.3 min |
| Enable MDT  | 1.1h |
|Enable MDT + current PR| 7s |

Disable MDT
![image](https://user-images.githubusercontent.com/69956021/218418777-a14b94ba-302c-483b-b7a8-401c45b62fc0.png)

Enable MDT
![image](https://user-images.githubusercontent.com/69956021/218418908-8f17c673-5102-426c-b0d9-a8f7f935ffe2.png)


Enable MDT + current PR
![image](https://user-images.githubusercontent.com/69956021/218418593-8b1a3b90-198e-4c77-9e1d-8fd67e61e6ce.png)


CC @yihua and @nsivabalan

Reply:
I think the idea of this PR is similar to https://github.com/apache/hudi/pull/7690, maybe you can refer to its conditions?
![image](https://user-images.githubusercontent.com/37108074/218666353-749c4d94-3112-493b-b0cc-69ecccc96d05.png)

Reply:
+1 to @Zouxxyy comment. For cleaner, we would need to load old file slices too (not just latest file-slices).